TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -457,18 +451,27 @@ inline void advance() {
  ProjectionRouterBuffer<VMProjType, AllProjectionType> projbuffer__, projbuffer____, projbuffer___;
  ap_uint<kNBits_MemAddr> projseq__, projseq___, projseq____;
 
-enum StubZPositionBarrelConsistency {
-  kBarrelPSMax = 1,
-  kBarrel2SMax = 5
-};
- enum StubZPositionDiskConsistency {
-   kDiskPSMax = 1,
-   kDisk2SMax = 3
-};
- enum StubPhiPositionConsistency {
-   kMax = 3
+ enum StubZPositionBarrelConsistency {
+   kBarrelPSMax = 1,
+   kBarrel2SMax = 5
+ };
+  enum StubZPositionDiskConsistency {
+    kDiskPSMax = 1,
+    kDisk2SMax = 3
+ };
+  enum StubPhiPositionConsistency {
+    kMax = 3
  };
 
+ static constexpr int projfinephibits = VMProjectionBase<VMProjType>::kVMProjFinePhiWideSize;
+ static constexpr int projfinezbits = VMProjectionBase<VMProjType>::kVMProjFineZSize;
+ static constexpr int stubfinephibits = VMStubMECMBase<VMSMEType>::kVMSMEFinePhiSize;
+ static constexpr int stubfinezbits = VMStubMECMBase<VMSMEType>::kVMSMEFineZSize;
+ const ap_uint<1 << 2*projfinephibits> phiLUT = isLessThanSize<projfinephibits,StubPhiPositionConsistency::kMax,false,projfinephibits,stubfinephibits>();
```

Comment:
To make it clear to the compiler that constants are known at compile time, I find it necessary to use "static const" (in cases where constexpr unavailable). Are you able to add the "static" keyword? e.g. https://stackoverflow.com/questions/34026555/how-to-define-a-static-const-variable-of-a-template-class .

Reply:
Using `static const` was causing issues in the HLS, but I can look a bit more into it.

Reply:
Ok, I've moved the definitions back into the function, which allows `static const` to be used.

Reply:
Should we go ahead and merge this, or wait for something to rebase the Future SW first to test?

Reply:
Yes, let's merge it.